pilot/pkg/networking/core/v1alpha3/gateway.go

```diff
@@ -144,6 +145,7 @@ func (configgen *ConfigGeneratorImpl) buildGatewayListeners(builder *ListenerBui
 			}
 
 			for cnum := range mutable.FilterChains {
+				mutable.FilterChains[cnum].TCP = append(mutable.FilterChains[cnum].TCP, xdsfilters.IstioNetworkAuthenticationFilter)
```

Comment:
Just confirming for my understanding - this is adding to GW. GW is often not doing mTLS. But that is fine, since this just sets filter state, and if its not mTLS it just doesn't set anything.

is that accurate

Reply:
It has no effect on data path.